core/domain/exp_jobs_one_off.py

```diff
@@ -529,6 +529,80 @@ def reduce(key, values):
         yield (key, values)
 
 
+class MathExplorationsImageGenerationAuditJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """Job that finds all the explorations with math rich text components and
+    creates a temporary storage model with all the information required for
+    generating math rich text component SVG images.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [exp_models.ExplorationModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        exploration = exp_fetchers.get_exploration_from_model(item)
+        size_of_math_svg = 0
+        largest_math_expression = ''
+        for state in exploration.states.values():
+            html_string = ''.join(state.get_all_html_content_strings())
+            size_of_math_svg_in_state = (
+                (html_validation_service.
+                 estimate_size_of_svg_for_math_expressions_in_html(
+                     html_string))[0])
+            largest_math_expression_in_state = (
+                (html_validation_service.
+                 estimate_size_of_svg_for_math_expressions_in_html(
+                     html_string))[1])
+            if (len(largest_math_expression_in_state) >
+                    len(largest_math_expression)):
+                largest_math_expression = largest_math_expression_in_state
+            size_of_math_svg = size_of_math_svg + size_of_math_svg_in_state
+        value_dict = {
+            'exploration_id': item.id,
+            'size_of_math_svgs': size_of_math_svg,
+            'largest_math_expression': largest_math_expression
+        }
+        yield ('Information about math-expressions in explorations', value_dict)
+
+    @staticmethod
+    def reduce(key, values):
+        final_values = [ast.literal_eval(value) for value in values]
+        number_of_explorations_having_math = 0
+        largest_math_expression = ''
+        estimated_no_of_batches = 1
+        size_of_math_svgs_in_a_batch = 0
+        for value in final_values:
+            if int(value['size_of_math_svgs']) > 0:
+                exp_models.MathExplorationImagesModel(
+                    id=value['exploration_id'],
+                    estimated_size_of_images_in_bytes=int(
+                        value['size_of_math_svgs'])).put()
+                number_of_explorations_having_math = (
+                    number_of_explorations_having_math + 1)
+                if (len(value['largest_math_expression']) >
+                        len(largest_math_expression)):
+                    largest_math_expression = value['largest_math_expression']
+                size_of_math_svgs_in_a_batch = (
+                    size_of_math_svgs_in_a_batch +
+                    int(value['size_of_math_svgs']))
+                if (size_of_math_svgs_in_a_batch >
+                        feconf.MAX_SIZE_OF_MATH_SVG_BATCH_BYTES):
```

Comment:
Define this constant in this file

Reply:
We will be using this limit in the future. So I think this might be the better place to define this